File: lib/ios/Modules/OfflineModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(AMapOffline, NSObject)

RCT_EXTERN_METHOD(download: (NSString)apiKey)
RCT_EXTERN_METHOD(addListener: (NSString)eventType)
RCT_EXTERN_METHOD(removeListeners: (int)count)

- (NSArray<NSString *> *)supportedEvents {
  return @[ @"download" ];
}

@end

File: lib/ios/Modules/OfflineModule.swift
@objc(AMapOffline)
class Offline: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool {
    false
  }

  @objc func removeListeners(_: Int) {}
  @objc func addListener(_: String) {}
  @objc func download(_: String) {}
}
